x-pack/plugins/timelines/server/search_strategy/timeline/index.ts

```diff
@@ -5,48 +5,165 @@
-        }),
-        mergeMap((esSearchRes) => queryFactory.parse(request, esSearchRes))
-      );
+
+      const queryFactory: TimelineFactory<T> = timelineFactory[factoryQueryType];
+
+      if (entityType != null && entityType === EntityType.ALERTS) {
+        return timelineAlertsSearchStrategy({
+          es: esAsInternal,
+          request,
+          options,
+          deps,
+          queryFactory,
+          alerting,
+          alertConsumers: alertConsumers ?? [],
+        });
+      } else {
+        return timelineSearchStrategy({ es, request, options, deps, queryFactory });
+      }
     },
     cancel: async (id, options, deps) => {
       if (es.cancel) {
         return es.cancel(id, options, deps);
+      } else if (esAsInternal.cancel) {
+        return esAsInternal.cancel(id, options, deps);
       }
     },
   };
 };
+
+const timelineSearchStrategy = <T extends TimelineFactoryQueryTypes>({
+  es,
+  request,
+  options,
+  deps,
+  queryFactory,
+}: {
+  es: ISearchStrategy;
+  request: TimelineStrategyRequestType<T>;
+  options: ISearchOptions;
+  deps: SearchStrategyDependencies;
+  queryFactory: TimelineFactory<T>;
+}) => {
+  const { entityType, alertConsumers, ...rest } = request;
+  const dsl = queryFactory.buildDsl(request);
+  return es.search({ request: { ...rest }, params: dsl }, options, deps).pipe(
+    map((response) => {
+      return {
+        ...response,
+        ...{
+          rawResponse: shimHitsTotal(response.rawResponse, options),
+        },
+      };
+    }),
+    mergeMap((esSearchRes) => queryFactory.parse(request, esSearchRes))
+  );
+};
+
+const timelineAlertsSearchStrategy = <T extends TimelineFactoryQueryTypes>({
+  es,
+  request,
+  options,
+  deps,
+  queryFactory,
+  alerting,
+  alertConsumers,
+}: {
+  es: ISearchStrategy;
+  request: TimelineStrategyRequestType<T>;
+  options: ISearchOptions;
+  deps: SearchStrategyDependencies;
+  alerting: AlertPluginStartContract;
+  queryFactory: TimelineFactory<T>;
+  alertConsumers: ALERTS_CONSUMERS[];
+}) => {
+  const allFeatureIdsValid = alertConsumers.every((id) => isValidFeatureId(id));
+
+  if (!allFeatureIdsValid) {
+    throw new Error('An invalid alerts consumer feature id was provided');
+  }
+
+  const indices = alertConsumers.flatMap((consumer) => mapConsumerToIndexName[consumer]);
+  const requestWithAlertsIndices = { ...request, defaultIndex: indices };
+
+  // Note: Alerts RBAC are built off of the alerting's authorization class, which
+  // is why we are pulling from alerting, not ther alertsClient here
+  const alertingAuthorizationClient = alerting.getAlertingAuthorizationWithRequest(deps.request);
+
+  const getAuthFilter = async () =>
+    alertingAuthorizationClient.getFindAuthorizationFilter(AlertingAuthorizationEntity.Alert, {
+      type: AlertingAuthorizationFilterType.ESDSL,
+      fieldNames: {
+        consumer: 'kibana.rac.alert.owner',
+        ruleTypeId: 'rule.id',
+        spaceIds: 'kibana.space_ids',
```

Comment:
Can we import these fields from the technical rule data field mapping in the `@kbn-rule-data-utils` package?